lib/eu_central_bank.rb

```diff
@@ -2,72 +2,64 @@
 require 'nokogiri'
 require 'money'
 require 'money/rates_store/store_with_historical_data_support'
-
-class InvalidCache < StandardError ; end
-
-class CurrencyUnavailable < StandardError; end
+require 'eu_central_bank/errors'
+require 'eu_central_bank/xml_parser'
 
 class EuCentralBank < Money::Bank::VariableExchange
-
   attr_accessor :last_updated
   attr_accessor :rates_updated_at
-  attr_accessor :historical_last_updated
-  attr_accessor :historical_rates_updated_at
 
-  SERIALIZER_DATE_SEPARATOR = '_AT_'
-
-  CURRENCIES = %w(USD JPY BGN CZK DKK GBP HUF ILS ISK PLN RON SEK CHF NOK HRK RUB TRY AUD BRL CAD CNY HKD IDR INR KRW MXN MYR NZD PHP SGD THB ZAR).map(&:freeze).freeze
+  CURRENCIES = %w[USD JPY BGN CZK DKK GBP HUF ILS ISK PLN RON SEK CHF NOK HRK RUB TRY AUD BRL CAD CNY HKD IDR INR KRW MXN MYR NZD PHP SGD THB ZAR].map(&:freeze).freeze
   ECB_RATES_URL = 'https://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml'.freeze
   ECB_90_DAY_URL = 'https://www.ecb.europa.eu/stats/eurofxref/eurofxref-hist-90d.xml'.freeze
+  ECB_ALL_URL = 'https://www.ecb.europa.eu/stats/eurofxref/eurofxref-hist.xml'.freeze
 
-  def initialize(st = Money::RatesStore::StoreWithHistoricalDataSupport.new, &block)
-    super
-    @currency_string = nil
-  end
-
-  def update_rates(cache=nil)
-    update_parsed_rates(doc(cache))
-  end
+  def initialize(store = Money::RatesStore::StoreWithHistoricalDataSupport.new, currencies: EuCentralBank::CURRENCIES, &block)
```

Comment:
That line is quite long. How about using `nil` as defaults and then doing `store || ...` and `currencies || ...`?